src/tests/network-api-test/network-api-test.c

```diff
@@ -0,0 +1,592 @@
+
+    /* OS_SocketGetIdByName */
+    expected = OS_INVALID_POINTER; 
+    actual = OS_SocketGetIdByName(NULL,NULL);
+    UtAssert_True(actual == expected, "OS_SocketGetIdByName() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_NAME_NOT_FOUND; 
+    actual = OS_SocketGetIdByName(&objid,"NF");
+    UtAssert_True(actual == expected, "OS_SocketGetIdByName() (%ld) == OS_ERR_NAME_NOT_FOUND", (long)actual);
+
+    /* OS_SocketGetInfo */
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketGetInfo(1, NULL);
+    UtAssert_True(actual == expected, "OS_SocketGetInfo() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_INVALID_ID;
+    actual = OS_SocketGetInfo(0, &prop);
+    UtAssert_True(actual == expected, "OS_SocketGetInfo() (%ld) == OS_ERR_INVALID_ID", (long)actual);
+
+} /* end TestDatagramNetworkApi */
+
+/*****************************************************************************
+ *
+ * Datagram Network Teardown
+ *
+ *****************************************************************************/
+void TestDatagramNetworkApi_Teardown(void)
+{
+    /* Close sockets */
+    OS_close(p1_socket_id);
+    OS_close(p2_socket_id);
+
+} /* end TestDatagramNetworkApi_Teardown */
+
+
+/*****************************************************************************
+ *
+ * Server Function for Stream Network Functional Test
+ *
+ *****************************************************************************/
+void Server_Fn(void)
+{
+    OS_SockAddr_t addr;
+    char Buf_send_s[4] = {0};
+    char Buf_rcv_s[4] = {0};
+    char Buf_expected[4] ={0};
+    int32 status;
+    uint32 connsock_id = 0;
+
+    /* Accept incoming connections */
+    status = OS_SUCCESS;
+    actual = OS_SocketAccept(s_socket_id, &connsock_id, &addr, OS_PEND);
+    UtAssert_True(status == expected, "OS_SocketAccept() (%ld) == OS_SUCCESS", (long)actual);
+    UtAssert_True(connsock_id != 0, "connsock_id (%lu) != 0", (unsigned long)connsock_id);
+
+    /* Once connected, send data to client */
+    strcpy(Buf_send_s, "abc");
+    status = sizeof(Buf_send_s);
+    actual = OS_TimedWrite(connsock_id, Buf_send_s, sizeof(Buf_send_s), OS_PEND);
+    UtAssert_True(actual == status, "OS_TimedWrite() (%ld) == %ld",(long)actual, (long)status);
+
+   /* Recieve incoming data from client*/
+    status = sizeof(Buf_rcv_s);
+    strcpy(Buf_expected, "xyz");
+
+    actual = OS_TimedRead(connsock_id, Buf_rcv_s, sizeof(Buf_rcv_s), OS_PEND);
+    UtAssert_True(actual == status, "OS_TimedRead() (%ld) == %ld",(long)actual, (long)status);
+    UtAssert_True(strcmp(Buf_rcv_s,Buf_expected) == 0, "Buf (%s) == Buf_expected (%s)", Buf_rcv_s, Buf_expected);
+
+    /*
+     * Test for invalid input parameters 
+     * to OS_SocketAccept
+     */
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketAccept(1, NULL, NULL, 0);
+    UtAssert_True(actual == expected, "OS_SocketAccept() (%ld) == OS_INVALID_POINTER", (long)actual);
```

Comment:
Note that these accept tests do not need to be done in a child thread, they can be done in the main test thread as they don't block.